$NetBSD: patch-gpr_gpr.gpr,v 1.3 2024/05/07 11:00:00 dkazankov Exp $

Fix name conflict with grpc - same as linux distros
Fix RELRO

--- gpr/gpr.gpr.orig	2023-10-09 21:31:11.000000000 +0300
+++ gpr/gpr.gpr	2024-05-15 09:39:04.188652027 +0300
@@ -38,7 +38,7 @@
 
    for Source_Dirs use ("src");
 
-   for Library_Name use "gpr";
+   for Library_Name use "gnatprj";
    for Object_Dir use "libobj/" & Bld & "/" & Gnat_Lib_Type;
    for Library_Dir use "lib/" & Bld & "/" & Gnat_Lib_Type;
 
@@ -73,6 +73,13 @@
             for Switches ("gpr*.ad?") use
               Compiler'Default_Switches ("Ada") & ("-g1");
       end case;
+      case Gnat_Lib_Type is
+         when "static" =>
+           null;
+         when "static-pic" | "relocatable" =>
+           for PIC_Option("Ada") use ("-fPIC");
+           for PIC_Option("C") use ("-fPIC");
+      end case;
    end Compiler;
 
    ------------
